src/main/java/rest/Notes.java

```diff
@@ -1,73 +1,72 @@
-import javax.ws.rs.core.Response;
-
-import io.quarkus.qute.RawString;
-import io.smallrye.common.annotation.Blocking;
-import io.smallrye.mutiny.Uni;
-import org.jboss.resteasy.reactive.RestForm;
-
-import io.quarkus.qute.TemplateInstance;
-import io.quarkus.qute.CheckedTemplate;
-import model.Note;
+import java.util.Date;
+import java.util.List;
 
 @Blocking
+@Path("/notes")
 public class Notes extends HxController {
 
     @CheckedTemplate
     public static class Templates {
-        public static native TemplateInstance notes(List<Note> notes, Long currentNoteId, Note currentNote);
+        public static native TemplateInstance notes(List<Note> notes, Long currentNoteId, Note currentNote, String search);
         public static native TemplateInstance notes$noteList(List<Note> notes, Long currentNoteId);
         public static native TemplateInstance notes$noteForm(Note currentNote);
     }
 
-    @Path("/")
-    public TemplateInstance notes(@QueryParam("id") Long id) {
-        final List<Note> notes = Note.listAllSortedByLastUpdated();
+    @Path("")
+    public TemplateInstance notes(@QueryParam("id") Long id, @QueryParam("search") String search) {
+        final List<Note> notes = Note.search(search);
         if (isHxRequest()) {
             return Templates.notes$noteList(notes, id);
         }
         Note note = id != null ? Note.findById(id) : null;
-        return Templates.notes(notes, id, note);
+        return Templates.notes(notes, id, note, search);
     }
 
-    @Path("/new-note")
     public TemplateInstance newNote() {
         // not really used
         Note note = new Note();
         if (isHxRequest()) {
-            return Templates.notes$noteForm(note);
+            this.hx(HxResponseHeader.TRIGGER, "refreshNoteList");
+            return  concatTemplates(
+                    Templates.notes$noteList(Note.listAllSortedByLastUpdated(), null),
+                    Templates.notes$noteForm(note)
+            );
         }
-        return Templates.notes(Note.listAllSortedByLastUpdated(), null, note);
+        return Templates.notes(Note.listAllSortedByLastUpdated(), null, note, null);
     }
 
-    @Path("/note/{id}")
+    @Path("{id}")
     public TemplateInstance editNote(@PathParam("id") Long id) {
         final Note note = Note.findById(id);
         if(note == null) {
-            notes(null);
+            notes(null, null);
             return null;
         }
         if (isHxRequest()) {
+            this.hx(HxResponseHeader.TRIGGER, "refreshNoteList");
             return  concatTemplates(
                     Templates.notes$noteList(Note.listAllSortedByLastUpdated(), id),
                     Templates.notes$noteForm(note)
                     );
         }
-        return Templates.notes(Note.listAllSortedByLastUpdated(), id, note);
+        return Templates.notes(Note.listAllSortedByLastUpdated(), id, note, null);
     }
 
-    @Path("/note/{id}/delete")
+    @Path("{id}")
```

Comment:
Pretty sure this `@Path("{id}")` is optional now, you can remove it :)

Yes: https://quarkiverse.github.io/quarkiverse-docs/quarkus-renarde/dev/index.html#_methods

Reply:
Right awesome